presto-spi/src/main/java/com/facebook/presto/spi/block/PageEncodingBuffers.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.block;
+
+import com.facebook.presto.spi.Page;
+import io.airlift.slice.BasicSliceOutput;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceOutput;
+import io.airlift.slice.Slices;
+
+import static com.facebook.presto.spi.block.BlockEncodingBuffersFactory.createBlockEncodingBuffers;
+import static com.facebook.presto.spi.block.PageBuilderStatus.DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
+import static io.airlift.slice.SizeOf.SIZE_OF_INT;
+import static java.lang.Math.max;
+
+public class PageEncodingBuffers
+{
+    private int channelCount;
+    int bufferedRowCount;
+    BlockEncodingBuffers[] blockEncodingBuffers;
+
+    private int capacity;
+
+    public PageEncodingBuffers()
+    {
+        channelCount = 0;
+        bufferedRowCount = 0;
+        blockEncodingBuffers = null;
+        capacity = 0;
+    }
+
+    public boolean isEmpty()
+    {
+        return bufferedRowCount == 0;
+    }
+
+    public boolean isFull()
+    {
+        return capacity <= getSizeInBytes();
+    }
+
+    public void prepareBuffers(Page page)
+    {
+        capacity = DEFAULT_MAX_PAGE_SIZE_IN_BYTES;
```

Comment:
remove this

Reply:
@mbasmanova I don't quite understand. Could you elaborate why we need to remove this line?

Reply:
@yingsu00 capacity is always set to the same fixed value, hence, it is better to initialize it one and make capacity a final variable.